libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts

```diff
@@ -591,14 +596,35 @@ export class SDSAutocompleteSearchComponent implements ControlValueAccessor {
       this._changeDetectorRef.markForCheck();
       if (this.model.items.length === 0) {
         this.inputValue = '';
-      } else if (this.configuration && this.configuration.selectionMode === SelectionMode.SINGLE){
-          this.inputValue = this.getObjectValue(
-            this.model.items[0],
-            this.configuration.primaryTextField
-          );
+      } else if (this.configuration && this.configuration.selectionMode === SelectionMode.SINGLE) {
+        this.inputValue = this.getObjectValue(
+          this.model.items[0],
+          this.configuration.primaryTextField
+        );
       }
     }
   }
+
+  addListener() {
+    let id = this.configuration.id;
+    let inputHeight = document.getElementById(id).getBoundingClientRect().height;
+    let modalHeight = document.getElementsByClassName('sds-dialog-content')[0].getBoundingClientRect().height;
+    document.getElementsByClassName('sds-dialog-content')[0]
+      .addEventListener('scroll', function (event) {
+        if ((document.getElementsByClassName('sds-autocomplete')).length > 0) {
+          let inputTopValue = document.getElementById(id).getBoundingClientRect().top;
+          let inputBottomValue = modalHeight - inputTopValue;
+          let listHeight = document.getElementsByClassName('sds-autocomplete')[0].getBoundingClientRect().height;
+          let element = document.getElementsByClassName('sds-autocomplete')[0].parentElement;
```

Comment:
Can we save some of these DOM queries to a variable so that we don't need to do multiple searches for the same element.
`document.getElementbyId(id)`
`document.getElementsByClassName('sds-dialog-content')`
`document.getElementsByClassName('sds-autocomplete')`

Reply:
Updated variables